litestar/_openapi/schema_generation/schema.py

```diff
@@ -363,11 +359,40 @@ def for_field_definition(self, field_definition: FieldDefinition) -> Schema | Re
             result = self.for_constrained_field(field_definition)
         elif field_definition.inner_types and not field_definition.is_generic:
             result = self.for_object_type(field_definition)
+        elif field_definition.is_subclass_of(UploadFile):
+            result = self.for_upload_file(field_definition)
         else:
             result = create_schema_for_annotation(field_definition.annotation)
 
         return self.process_schema_result(field_definition, result) if isinstance(result, Schema) else result
 
+    def for_upload_file(self, field_definition: FieldDefinition) -> Schema:
+        """Create schema for UploadFile.
+
+        Args:
+            field_definition: A field definition instance.
+
+        Returns:
+            A Schema instance.
+        """
+
+        if field_definition.is_mapping:
+            return Schema()
```

Comment:
I think the correct solution is to just give return the same schema that we would if it was `list[UploadFile]` since that's what we're really expecting from the client. It's just that we convert that into a dictionary internally for ease of use. Any thoughts on this?